### PR TITLE
Fix base exception message

### PR DESCRIPTION
**Pull request by wbingli**
*Issue #, if available:*  #243 

*Description of changes:*. Fix the error message issue for BaseHandlerException. 

Cloudformation service doesn't like empty message exception, use the cause exception as default exception message.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by miparnisari** [comment]
So this is why this code

```java
try {
            awsResponse = proxyClient.injectCredentialsAndInvokeV2(awsRequest, proxyClient.client()::deleteMetricFilter);
        } catch (ResourceNotFoundException e) {
            logger.log("Resource does not exist and could not be deleted.");
            throw new CfnNotFoundException(e);
        }
```

causes this to be printed in the handler logs...

```

  | 2020-05-15T15:00:53.943-07:00 | Resource does not exist and could not be deleted.
  | 2020-05-15T15:00:54.011-07:00 | null in a DELETE action on a AWS::Logs::MetricFilter: software.amazon.cloudformation.exceptions.CfnNotFoundExceptionsoftware.amazon.cloudformation.exceptions.CfnNotFoundException	at software.amazon.logs.metricfilter.DeleteHandler.deleteResource(DeleteHandler.java:51)

File: src/main/java/software/amazon/cloudformation/exceptions/BaseHandlerException.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
     protected BaseHandlerException(final Throwable cause,
                                    final HandlerErrorCode errorCode) {
-        super(null, cause);
+        super(cause.getMessage(), cause);
```
Comment (aygold92) [inline src/main/java/software/amazon/cloudformation/exceptions/BaseHandlerException.java:29]: I think you can just call `super(cause)`

File: src/main/java/software/amazon/cloudformation/exceptions/CfnInvalidCredentialsException.java [hunk 2]
```diff
@@ -22,10 +22,10 @@
     private static final HandlerErrorCode ERROR_CODE = HandlerErrorCode.InvalidCredentials;
 
     public CfnInvalidCredentialsException() {
-        this(null);
+        super(ERROR_CODE.getMessage(), ERROR_CODE);
     }
 
     public CfnInvalidCredentialsException(final Throwable cause) {
```
Comment (aygold92) [inline src/main/java/software/amazon/cloudformation/exceptions/CfnInvalidCredentialsException.java:28]: I don't see any reason to limit all of these exceptions to just having a `cause` constructor.  Can we include a `(cause)` and a `(message, cause)` for all of them?

Comment (aygold92) [reply src/main/java/software/amazon/cloudformation/exceptions/CfnInvalidCredentialsException.java:28]: can we create an issue to resolve this then, if we're not going to include it in this PR?

Comment (wbingli) [reply src/main/java/software/amazon/cloudformation/exceptions/CfnInvalidCredentialsException.java:28]: Tracked in #276

**Pull request merged**